sdk/spring/spring-cloud-azure-core/src/main/java/com/azure/spring/cloud/core/implementation/credential/resolver/AzureTokenCredentialResolver.java

```diff
@@ -44,19 +44,24 @@ private static TokenCredential resolveTokenCredential(AzureProperties azurePrope
         final String tenantId = azureProperties.getProfile().getTenantId();
         final String clientId = properties.getClientId();
         final boolean isClientIdSet = StringUtils.hasText(clientId);
+        final String authorityHost = azureProperties.getProfile().getEnvironment().getActiveDirectoryEndpoint();
+
         if (StringUtils.hasText(tenantId)) {
 
             if (isClientIdSet && StringUtils.hasText(properties.getClientSecret())) {
                 return new ClientSecretCredentialBuilder().clientId(clientId)
+                                                          .authorityHost(authorityHost)
                                                           .clientSecret(properties.getClientSecret())
                                                           .tenantId(tenantId)
                                                           .build();
             }
 
             String clientCertificatePath = properties.getClientCertificatePath();
             if (StringUtils.hasText(clientCertificatePath)) {
-                ClientCertificateCredentialBuilder builder = new ClientCertificateCredentialBuilder().tenantId(tenantId)
-                                                                                                     .clientId(clientId);
+                ClientCertificateCredentialBuilder builder = new ClientCertificateCredentialBuilder()
```

Comment:
do we still need to configure authority host specifially if my pr for kafka oauth2 uses the credential builder factory here

Reply:
I don't think so.